src/MOI_wrapper.jl

```diff
@@ -38,7 +49,7 @@ end
 mutable struct Optimizer{OT <: MOI.ModelLike} <: MOI.AbstractOptimizer
     optimizer::OT
     primal_optimal::Array{Float64}  # solution
-    dual_optimal::Array{Float64}  
+    dual_optimal::Array{Union{Array{Float64}, Float64}}  # refer - https://github.com/AKS1996/DiffOpt.jl/issues/21#issuecomment-651130485
```

Comment:
Why not using `Vector` instead of `Array` in primal_ and dual_optimal?

Reply:
no preference :sweat_smile: will edit now

Reply:
Array sort-of indicates a multi-dimensional array, while vector is specifically one-dimensional